pkg/kubelet/kubelet.go

```diff
@@ -2310,22 +2313,55 @@ func (kl *Kubelet) rejectPod(pod *v1.Pod, reason, message string) {
 // the pod cannot be admitted.
 // allocatedPods should represent the pods that have already been admitted, along with their
 // admitted (allocated) resources.
-func (kl *Kubelet) canAdmitPod(allocatedPods []*v1.Pod, pod *v1.Pod) (bool, string, string) {
+func (kl *Kubelet) canAdmitPod(allocatedPods []*v1.Pod, pod *v1.Pod) (ok bool, reason string, msg string) {
+	defer func() {
+		if ok {
+			return
+		}
+
+		allowlist := []string{
+			lifecycle.AppArmorNotAdmittedReason,
+			lifecycle.PodOSSelectorNodeLabelDoesNotMatch,
+			lifecycle.PodOSNotSupported,
+			lifecycle.InvalidNodeInfo,
+			lifecycle.InitContainerRestartPolicyForbidden,
+			lifecycle.UnexpectedAdmissionError,
+			lifecycle.UnknownReason,
+			lifecycle.UnexpectedPredicateFailureType,
+			tainttoleration.ErrReasonNotMatch,
+			eviction.Reason,
+			sysctl.ForbiddenReason,
+			topologymanager.ErrorTopologyAffinity,
+		}
+		if runtime.GOOS == "linux" {
+			allowlist = append(allowlist, nodeshutdown.NodeShutdownNotAdmittedReason)
```

Comment:
You can just keep this in the allowlist, it doesn't matter if it's linux or not. If this reason is returned on Windows, we still want to know. The point of the allowlist is just to make sure we don't accidentally end up with a reason that includes a pod.UID and blow up the metrics storage.